src/components/Table/TableHead/TableHead.jsx

```diff
@@ -350,14 +354,44 @@ const TableHead = ({
                 'table-header-sortable': matchingColumnMeta.isSortable,
                 [`${iotPrefix}--table-header-resize`]: hasResize,
               })}
+              // data-floating-menu-container is a work around for this carbon issue: https://github.com/carbon-design-system/carbon/issues/4755
+              data-floating-menu-container
             >
               <TableCellRenderer
+                className={`${iotPrefix}--table-head--text`}
                 wrapText={wrapCellText}
                 truncateCellText={truncateCellText}
                 allowTooltip={false}
               >
                 {matchingColumnMeta.name}
               </TableCellRenderer>
+
+              {matchingColumnMeta.options !== undefined && matchingColumnMeta.options !== null ? (
+                <OverflowMenu
+                  className={`${iotPrefix}--table-head--overflow`}
+                  direction="bottom"
+                  data-testid="table-head--overflow"
+                  flipped={columnIndex !== 0}
+                  onClick={e => {
+                    e.stopPropagation();
```

Comment:
Can we remove this since we are stoping propagation in the `OverflowMenuItem`?

Reply:
if the header is clickable (if column sorting is available) the overflow menu item will still trigger the sort.